components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -262,19 +291,43 @@ std::optional<std::string> GetAnkrBlockchainFromChainId(
   return std::nullopt;
 }
 
-}  // namespace
+void MergeOwnedTokenAccounts(
+    JsonRpcService::GetSolanaTokenAccountsByOwnerCallback callback,
+    std::vector<SolanaRPCResponse<std::vector<SolanaAccountInfo>>> responses) {
+  std::vector<SolanaAccountInfo> token_accounts;
+  for (auto& response : responses) {
+    if (response.error != mojom::SolanaProviderError::kSuccess) {
+      std::move(callback).Run({}, response.error, response.error_message);
+      return;
+    }
+    token_accounts.insert(token_accounts.end(), response.value.begin(),
+                          response.value.end());
+  }
 
-struct PendingAddChainRequest {
-  mojom::AddChainRequestPtr request;
-  url::Origin origin;
-};
+  std::move(callback).Run(token_accounts, mojom::SolanaProviderError::kSuccess,
+                          "");
+}
 
-struct PendingSwitchChainRequest {
-  mojom::SwitchChainRequestPtr switch_chain_request;
-  url::Origin origin;
-  base::Value switch_chain_id;
-  mojom::EthereumProvider::RequestCallback switch_chain_callback;
-};
+void MergeGetSPLTokenBalances(
```

Comment:
Can we refactor `MergeGetSPLTokenBalances` and `MergeOwnedTokenAccounts` into one function using two template typename? Even though `SolanaAccountInfo` is copyable, it can also be moved.

Reply:
https://github.com/brave/brave-core/pull/23945/commits/b2cda274af90780bfb0b9e7a1195a9b74f549248